waku/peer.go

```diff
@@ -277,6 +313,21 @@ func (p *Peer) bloomMatch(env *Envelope) bool {
 	return p.fullNode || BloomFilterMatch(p.bloomFilter, env.Bloom())
 }
 
+func (p *Peer) topicInterestMatch(env *Envelope) bool {
+	p.topicInterestMu.Lock()
```

Comment:
What does the lock protect against? Perhaps a comment here would be useful

Reply:
Basically `map`s (or slices) are not thread safe, so we need to acquire a lock before accessing the datastructure, as they might get accessed them from different threads, I will leave a comment